File: src/main/java/com/leon/bugreport/Category.java
package com.leon.bugreport;

import org.bukkit.Material;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.meta.ItemMeta;

import java.util.Objects;

public class Category {
	private final int id;
	private final String color;
	private final String name;
	private final ItemStack itemStack;

	Category(int id, String name, String color, ItemStack itemStack) {
		this.id = id;
		this.name = name;
		this.color = color;
		this.itemStack = itemStack;
	}

	public int getId() {
		return id;
	}

	public String getColor() {
		return color;
	}

	public String getName() {
		return name;
	}

	public String getDescription() {
		ItemMeta itemMeta = itemStack.getItemMeta();
		if (itemMeta == null || !itemMeta.hasLore()) {
			return "";
		}
		return Objects.requireNonNull(itemMeta.getLore()).get(0);
	}

	public Material getItem() {
		return itemStack.getType();
	}
}

File: src/main/java/com/bugreportmc/bugreport/Category.kt
package com.bugreportmc.bugreport

import org.bukkit.Material
import org.bukkit.inventory.ItemStack
import java.util.*

class Category internal constructor(
	@JvmField val id: Int,
	@JvmField val name: String,
	@JvmField val color: String,
	private val itemStack: ItemStack
) {
	fun getName(): String {
		return name
	}

	val description: String
		get() {
			val itemMeta = itemStack.itemMeta
			if (itemMeta == null || !itemMeta.hasLore()) {
				return ""
			}
			return Objects.requireNonNull(itemMeta.lore)[0]
		}

	val item: Material
		get() = itemStack.type
}
